cmd/argo/commands/stop.go

```diff
@@ -33,29 +58,55 @@ func NewStopCommand() *cobra.Command {
   argo stop @latest
 `,
 		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 && !o.isList() {
+				cmd.HelpFunc()(cmd, args)
+				os.Exit(1)
+			}
 
 			ctx, apiClient := client.NewAPIClient()
 			serviceClient := apiClient.NewWorkflowServiceClient()
-			namespace := client.Namespace()
+			o.namespace = client.Namespace()
 
-			selector, err := fields.ParseSelector(stopArgs.nodeFieldSelector)
+			selector, err := fields.ParseSelector(o.nodeFieldSelector)
 			if err != nil {
-				log.Fatalf("Unable to parse node field selector '%s': %s", stopArgs.nodeFieldSelector, err)
+				log.Fatalf("Unable to parse node field selector '%s': %s", o.nodeFieldSelector, err)
+			}
+
+			var workflows wfv1.Workflows
+
+			if o.isList() {
+				listed, err := listWorkflows(ctx, serviceClient, listFlags{
+					namespace: o.namespace,
+					fields:    o.fields,
+					labels:    o.labels,
+				})
+				errors.CheckError(err)
+				workflows = append(workflows, listed...)
+			} else {
+				workflows = o.convertToWorkflows(args)
 			}
 
-			for _, name := range args {
+			for _, w := range workflows {
+				if o.dryRun {
+					fmt.Printf("workflow %s stopped (dry-run)\n", w.Name)
```

Comment:
I think in other commands we put quotes around the workflow name (maybe single or maybe double?)

Reply:
It's not; there are no quotes around the workflow name. (e.g. [terminate](https://github.com/argoproj/argo/blob/master/cmd/argo/commands/terminate.go#L82))